src/garage/torch/algos/vpg.py

```diff
@@ -193,15 +227,16 @@ def _compute_loss(self, itr, obs, actions, rewards, valids, baselines):
         advantages = compute_advantages(self.discount, self._gae_lambda,
                                         self.max_path_length, baselines,
                                         rewards)
-
         if self._center_adv:
-            means, variances = list(
-                zip(*[(valid_adv.mean(), valid_adv.var())
-                      for valid_adv in filter_valids(advantages, valids)]))
-            advantages = F.batch_norm(advantages.t(),
-                                      torch.Tensor(means),
-                                      torch.Tensor(variances),
-                                      eps=self._eps).t()
+            valid_transposed_adv = [
+                adv[valids > i] for i, adv in enumerate(advantages.T)
+            ]
+            means = torch.Tensor([adv.mean() for adv in valid_transposed_adv])
+            means[torch.isnan(means)] = 0
```

Comment:
is this an acceptable hotpatch? Are there some instabilities when the advantages are computed that have to be resolved by zeroing out NaNs when they appear in your computation path?

Reply:
i'm not comfortable merging this until someone can explain *why* it's necessary.

if we understand the root cause and this is an appropriate fix, sure.

please leave a large comment explaining what the root cause is here and why this fix is necessary.